File: dynamic_programming/python/121_best_time_to_buy_and_to_sell_stock.py
class solution:
    def maxProfit(self,prices:list[int])->int:
        min_price = prices[0]
        max_price = 0
        
        for i in range(len(prices)):
            if prices[i] < min_price:
                min_price = prices[i]
            elif prices[i] - min_price > max_price:
                max_price = prices[i] - min_price

        return max_price

File: dynamic programming/c/121_best_time_to_buy_and_to_sell_stock.c
int maxProfit(int *prices, int pricesSize){
    int dp_sell[pricesSize];
    int dp_buy[pricesSize];

    dp_sell[0]=0;
    dp_buy[0]=-prices[0];

    for(int i=0;i<pricesSize;i++){
        dp_sell[i] = dp_buy[i-1]+
    }


    return dp_sell[pricesSize-1];
}